File: app/src/main/java/in/example/rahul/vegcartpro/activity/OrderActivity.java
package in.example.rahul.vegcartpro.activity;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.iid.FirebaseInstanceId;

import java.text.NumberFormat;

import in.example.rahul.vegcartpro.Model.Cart;
import in.example.rahul.vegcartpro.R;

public class OrderActivity extends AppCompatActivity {
/*  onclick cart image on detail activity this will open
  * Use: To put order size and cost  */
    TextView tvQuantity,tvPrice;
    public double numberOfVeg=1 ;
    public double priceOfVeg ;
    public String foodprice;
    public String foodname;
    EditText address;
    private static final String TAG = "OrderActivity";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_order);
        tvQuantity=findViewById(R.id.quantity_text_view);
        tvPrice=findViewById(R.id.price_text_view);
        address=findViewById(R.id.et_add);
        Bundle bundle=getIntent().getExtras();
        foodprice= bundle.getString("pricefood");
        foodname=bundle.getString("namefood");
        priceOfVeg= Double.parseDouble(foodprice);
        tvPrice.setText("₹ "+String.valueOf(foodprice));
    }
    public void submitOrder(View v){

       // String mPrice="You order "+numberOfVeg+" kg of Vegetable \n Thank you !";
        //textView2.setText(String.valueOf(mPrice));
        String quantityVeg=tvQuantity.getText().toString();
        String priceVeg= tvPrice.getText().toString();
        String addVeg=address.getText().toString();
        //String nameVeg= ;
        if (address.equals("") || address.length()==0 ){
            Toast.makeText(getBaseContext(),"Please Enter Address \n Cant leave blank",Toast.LENGTH_SHORT).show();
        }
       else {
        final FirebaseDatabase database=FirebaseDatabase.getInstance();

        final DatabaseReference ref=database.getReference("Cart");

        DatabaseReference newPostRef=ref.push();

        newPostRef.setValue(new Cart(foodname,tvPrice.getText().toString(),tvQuantity.getText().toString(),address.getText().toString()));

        Toast.makeText(getBaseContext(),"Order Placed Successfully \n Thank you",Toast.LENGTH_SHORT).show();   }

        //Intent likeIntent = new Intent(this, LikeServices.class);
       // likeIntent.putExtra(NOTIFICATION_ID_EXTRA,notificationId);

        String token= FirebaseInstanceId.getInstance().getToken();
        Log.d(TAG,"Token: "+token);
        Toast.makeText(OrderActivity.this,token,Toast.LENGTH_SHORT).show();
    }
    public void increment(View v){
        numberOfVeg=numberOfVeg+0.5;
        // Increment of 500gm will done onclick + button
        display(numberOfVeg);
        displayPrice(numberOfVeg*priceOfVeg);
    }
    public void decrement(View v){
        if (numberOfVeg >= 1) {
            numberOfVeg=numberOfVeg-0.5;
        } else {
        }

        // Decrement of 500gm will done onclick - button
        display(numberOfVeg);
        displayPrice(numberOfVeg*priceOfVeg);
    }
    // display the quantity between + and - button
    private void display(double i) {
        tvQuantity.setText(""+i);
    }
    // display price onclick +,- button
    private void displayPrice(double number) {

        tvPrice.setText(NumberFormat.getCurrencyInstance().format(number));
    }
}

File: app/src/main/java/in/rahulkhatri/vegcartpro/activity/OrderActivity.kt
package `in`.rahulkhatri.vegcartpro.activity

import `in`.rahulkhatri.vegcartpro.model.CartModel
import `in`.rahulkhatri.vegcartpro.R
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.util.Log
import android.view.View
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.iid.FirebaseInstanceId
import java.text.NumberFormat

class OrderActivity : AppCompatActivity() {
    /*  onclick cart image on detail activity this will open
  * Use: To put order size and cost  */
    lateinit var tvQuantity: TextView
    lateinit var tvPrice: TextView
    var numberOfVeg = 1.0
    var priceOfVeg = 0.0
    var foodprice = ""
    var foodname = ""
    lateinit var address: EditText
    private val TAG = OrderActivity::class.java.simpleName
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_order)
        tvQuantity = findViewById(R.id.quantity_text_view)
        tvPrice = findViewById(R.id.price_text_view)
        address = findViewById(R.id.et_add)
        val bundle = intent.extras
        foodprice = bundle?.getString("pricefood")!!
        foodname = bundle.getString("namefood")!!
        priceOfVeg = foodprice.toDouble()
        tvPrice.text = "₹ $foodprice"
    }

    fun submitOrder(v: View?) { // String mPrice="You order "+numberOfVeg+" kg of Vegetable \n Thank you !";
//textView2.setText(String.valueOf(mPrice));
        val quantityVeg = tvQuantity.text.toString()
        val priceVeg = tvPrice.text.toString()
        val addVeg = address.text.toString()
        //String nameVeg= ;
        if (address.equals(" ") || address.length() == 0) {
            Toast.makeText(baseContext, "Please Enter Address \n Cant leave blank", Toast.LENGTH_SHORT).show()
        } else {
            val database = FirebaseDatabase.getInstance()
            val ref = database.getReference("Cart")
            val newPostRef = ref.push()
            newPostRef.setValue(CartModel(foodname, tvPrice.getText().toString(), tvQuantity.text.toString(), address.text.toString()))
            Toast.makeText(baseContext, "Order Placed Successfully \n Thank you", Toast.LENGTH_SHORT).show()
        }
        //Intent likeIntent = new Intent(this, LikeServices.class);
// likeIntent.putExtra(NOTIFICATION_ID_EXTRA,notificationId);
        val token = FirebaseInstanceId.getInstance().token
        Log.d(TAG, "Token: $token")
        Toast.makeText(this@OrderActivity, token, Toast.LENGTH_SHORT).show()
    }

    fun increment(v: View?) {
        numberOfVeg = numberOfVeg + 0.5
        // Increment of 500gm will done onclick + button
        display(numberOfVeg)
        displayPrice(numberOfVeg * priceOfVeg)
    }

    fun decrement(v: View?) {
        if (numberOfVeg >= 1) {
            numberOfVeg = numberOfVeg - 0.5
        }
        // Decrement of 500gm will done onclick - button
        display(numberOfVeg)
        displayPrice(numberOfVeg * priceOfVeg)
    }

    // display the quantity between + and - button
    private fun display(i: Double) {
        tvQuantity.text = "$i"
    }

    // display price onclick +,- button
    private fun displayPrice(number: Double) {
        tvPrice.text = NumberFormat.getCurrencyInstance().format(number)
    }
}